.dagger/docs.go

```diff
@@ -260,6 +264,116 @@ export const daggerVersion = "%s";
 	return dir, nil
 }
 
+// Deploys a current build of the docs.
+func (d Docs) Deploy(
+	ctx context.Context,
+	netlifyToken *dagger.Secret,
+) (string, error) {
+	commit, err := d.Dagger.Git.Head().Commit(ctx)
+	if err != nil {
+		return "", err
+	}
+	dirty, err := d.Dagger.Git.Dirty(ctx)
+	if err != nil {
+		return "", err
+	}
+	message := "Manual build on " + commit
+	if dirty {
+		message += "-dirty"
+	}
+
+	out, err := dag.Container().
+		From("node:18").
+		WithExec([]string{"npm", "install", "netlify-cli", "-g"}). // pin!!!!
+		WithEnvVariable("NETLIFY_SITE_ID", "docs-dagger-io").
+		WithSecretVariable("NETLIFY_AUTH_TOKEN", netlifyToken).
+		WithMountedDirectory("/build", d.Site()).
+		WithExec([]string{"netlify", "deploy", "--dir=/build", "--branch=main", "--message", message, "--json"}).
+		Stdout(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	var dt struct {
+		DeployID string `json:"deploy_id"`
+	}
+	if err := json.Unmarshal([]byte(out), &dt); err != nil {
+		return "", err
+	}
+
+	return dt.DeployID, nil
+}
+
+// Publish a previous deployment to production - defaults to the latest deployment on the main branch.
+func (d Docs) Publish(
+	ctx context.Context,
+	netlifyToken *dagger.Secret,
+	// +optional
+	deployment string,
+) error {
+	api := "https://api.netlify.com/api/v1"
+	site := "docs.dagger.io"
+	branch := "main"
+	client := http.Client{}
+
+	token, err := netlifyToken.Plaintext(ctx)
```

Comment:
Rather than adding this as a GitHub Secret, what are your thoughts @jedevc on wiring this through with `onepassword-sdk-go`, same as we do in https://github.com/dagger/dagger.io/blob/1408b23aaa0c3744e6a1b3831d97d6f452ef6408/dagviz/.dagger/main.go#L51-L83 ?

cc @aluzzardi

Reply:
Ideally, I'd like to avoid putting the onepassword integration into our CI code itself - it makes sense for our dagger/dagger.io repo, since it's private and so can do things like depend on the structure of the vault. But, we probably shouldn't do that for our public CI IMO.

I'd like to use the secret providers when they land - that feels like the right way to go. Especially as we'll want to do that conversion *anyways* - if we integrate `onepassword-sdk-go`, we'll probably be ripping it out later to use the secret providers anyways, so would rather just hold on this action for now.